drivers/wireless/ieee80211/bcm43xxx/bcmf_sdpcm.c

```diff
@@ -141,102 +141,75 @@ int bcmf_sdpcm_readframe(FAR struct bcmf_dev_s *priv)
   uint16_t len;
   uint16_t checksum;
   struct bcmf_sdpcm_header *header;
-  struct bcmf_sdpcm_header tmp_hdr;
   struct bcmf_sdio_frame *sframe;
   FAR struct bcmf_sdio_dev_s *sbus = (FAR struct bcmf_sdio_dev_s *)priv->bus;
 
+  /* Request free frame buffer */
+
+  sframe = bcmf_sdio_allocate_frame(priv, false, false);
```

Comment:
why not blocking

Reply:
nonblocking alloc before read sdpcm is more reasonable, driver should not discard the data